### PR TITLE
Fix and improve report script

### PR DESCRIPTION
**Pull request by fvaleri**
- Fix: command not found error with namespaced operator/drain-cleaner
- Fix: missing pods and logs when using --bridge option
- Improvement: add short descriptions to the usage prompt
- Improvement: merge CRDs and CRs download blocks

## Command not found issue

Happens with namespaced CO deployments.

```sh
$ tools/report.sh --namespace=test --cluster=my-cluster --out-dir=~/Downloads
...
clusterrolebindings
    clusterrolebinding.rbac.authorization.k8s.io/strimzi-cluster-operator
    clusterrolebinding.rbac.authorization.k8s.io/strimzi-cluster-operator-kafka-broker-delegation
    clusterrolebinding.rbac.authorization.k8s.io/strimzi-cluster-operator-kafka-client-delegation
clusteroperator
    deployment.apps/strimzi-cluster-operator
    pod/strimzi-cluster-operator-77448fbdd6-v2c5h
tools/report-old.sh: line 229: get_pod_logs: command not found
```

## Bridge issue

Happens always.

```sh
### BEFORE
$ tools/report.sh --namespace=test --cluster=my-cluster --bridge=my-bridge --out-dir=~/Downloads
pods
    pod/my-cluster-entity-operator-6c56d7f4fb-l874k
    pod/my-cluster-kafka-0
    pod/my-cluster-kafka-1
    pod/my-cluster-kafka-2
    pod/my-cluster-zookeeper-0
    pod/my-cluster-zookeeper-1
    pod/my-cluster-zookeeper-2
...
podlogs
    my-cluster-entity-operator-6c56d7f4fb-l874k
    my-cluster-kafka-0
    my-cluster-kafka-1
    my-cluster-kafka-2
    my-cluster-zookeeper-0
    my-cluster-zookeeper-1
    my-cluster-zookeeper-2
...

### AFTER
$ tools/report.sh --namespace=test --cluster=my-cluster --bridge=my-bridge --out-dir=~/Downloads
...
pods
    pod/my-cluster-entity-operator-6c56d7f4fb-l874k
    pod/my-cluster-kafka-0
    pod/my-cluster-kafka-1
    pod/my-cluster-kafka-2
    pod/my-cluster-zookeeper-0
    pod/my-cluster-zookeeper-1
    pod/my-cluster-zookeeper-2
    pod/my-bridge-bridge-5d8f74f9b7-fntzv <<<<<<<
...
podlogs
    my-cluster-entity-operator-6c56d7f4fb-l874k
    my-cluster-kafka-0
    my-cluster-kafka-1
    my-cluster-kafka-2
    my-cluster-zookeeper-0
    my-cluster-zookeeper-1
    my-cluster-zookeeper-2
    my-bridge-bridge-5d8f74f9b7-fntzv <<<<<<<<
...
```

## Improved help

```sh
### BEFORE
$ tools/report.sh
Usage: report.sh [required] [optional]

Required:
  --namespace=<string>
  --cluster=<string>

Optional:
  --bridge=<string>
  --connect=<string>
  --mm2=<string>
  --out-dir=<string>
  --secrets=(off|hidden|all)]

Default level of secret verbosity is 'hidden' (only secret key will be reported).

### AFTER
$ tools/report.sh
Usage: report.sh [options]

Required:
  --namespace=<string>          Kubernetes namespace.
  --cluster=<string>            Kafka cluster name.

Optional:
  --bridge=<string>             Bridge component name to get pods and logs.
  --connect=<string>            Connect component name to get pods and logs.
  --mm2=<string>                MM2 component name to get pods and logs.
  --secrets=(off|hidden|all)    Secret verbosity. Default is hidden, only the secret key will be reported.
  --out-dir=<string>            Script output directory.
```

### CODE REVIEW

File: tools/report.sh [hunk 1]
```diff
@@ -287,35 +284,9 @@ if [[ -n $EVENTS ]]; then
...
-}
-
 echo "podlogs"
-mkdir -p "$OUT_DIR"/reports/podlogs
-mkdir -p "$OUT_DIR"/reports/configs
 PODS=$($KUBE_CLIENT get po -l strimzi.io/cluster="$CLUSTER" -o name -n "$NAMESPACE" | cut -d "/" -f 2)
-PODS="$PODS $($KUBE_CLIENT get po -l strimzi.io/cluster="$BRIDGE" -o name -n "$NAMESPACE" | cut -d "/" -f 2)"
-PODS="$PODS $($KUBE_CLIENT get po -l strimzi.io/cluster="$CONNECT" -o name -n "$NAMESPACE" | cut -d "/" -f 2)"
-PODS="$PODS $($KUBE_CLIENT get po -l strimzi.io/cluster="$MM2" -o name -n "$NAMESPACE" | cut -d "/" -f 2)"
+PODS="$PODS $($KUBE_CLIENT get po -l strimzi.io/cluster="$COMP_NAME" -o name -n "$NAMESPACE" | cut -d "/" -f 2)"
```
Comment (scholzj) [inline tools/report.sh:289]: This expects only one comopenent name and that does not seem to make much sense to me. As this is currently centered around the Kafka cluster (which does not make sense, but that is another story), it is natural that you might want to collect the Kafka cluster together with Bridge, Connect and MM2. Not just one of them.

File: tools/report.sh [hunk 2]
```diff
@@ -216,6 +202,29 @@ for RES in "${CLUSTER_RESOURCES[@]}"; do
...
+    fi
+    if [[ "$count" -gt 1 && -n "$con" && "$names" == *"$con"* ]]; then
+      logs="$($KUBE_CLIENT -n "$NAMESPACE" logs "$pod" -c "$con" ||true)"
+      if [[ -n $logs ]]; then printf "%s" "$logs" > "$OUT_DIR"/reports/podlogs/"$pod"-"$con".log; fi
+      logs="$($KUBE_CLIENT -n "$NAMESPACE" logs "$pod" -p -c "$con" 2>/dev/null ||true)"
+      if [[ -n $logs ]]; then printf "%s" "$logs" > "$OUT_DIR"/reports/podlogs/"$pod"-"$con".log.0; fi
+    fi
+  fi
+}
+
```
[the remark this answers is not in the document]
Comment (fvaleri) [reply tools/report.sh:227]: The function needs to be declared before it is first used, that's why I moved. The only change to this function is that I'm now creating the required directories inside the function (line 212), which is safer.

File: tools/report.sh (hunk 1 above)
Comment (fvaleri) [reply tools/report.sh:289]: Right, going to revert this change. Thanks.

File: tools/report.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply tools/report.sh:237]: You do not seem to use the configs directory there. so why should you create it there?

**Review by scholzj — APPROVED** [review]
LGTM assuming it works.

**Review by ppatierno — APPROVED** [review]
Regarding the improvements, they seem to be be useful.
Also thanks for fixing the bridge bug.
LGTM.

**Pull request merged**